snapcraft/lifecycle.py

```diff
@@ -226,19 +226,71 @@ def _reverse_dependency_tree(config, part_name):
     return dependents
 
 
-def _clean_part_and_all_dependents(config, part, staged_state, stripped_state,
-                                   step):
-    # Clean the part in question
-    part.clean(staged_state, stripped_state, step)
-
-    # Now obtain the reverse dependency tree for this part. Make sure
-    # all dependents are also cleaned.
-    dependents = _reverse_dependency_tree(config, part.name)
+def _clean_part_and_all_dependents(part_name, step, config, staged_state,
+                                   stripped_state):
+    # Obtain the reverse dependency tree for this part. Make sure all
+    # dependents are cleaned.
+    dependents = _reverse_dependency_tree(config, part_name)
     dependent_parts = {p for p in config.all_parts
                        if p.name in dependents}
     for dependent_part in dependent_parts:
         dependent_part.clean(staged_state, stripped_state, step)
 
+    # Finally, clean the part in question
+    config.get_part(part_name).clean(staged_state, stripped_state, step)
+
+
+def _humanize_list(items):
```

Comment:
if monkeys ever read or if an extra terrestrial sees this they will feel left out :wink: